src/names.rs

```diff
@@ -28,6 +28,37 @@ impl Regex {
             _phantom: PhantomData,
         }
     }
+
+    /// Calls `callback` for each named group in the regex. Each callback gets the group name
+    /// and group indices.
+    pub fn foreach_name<F>(&self, mut callback: F) -> i32
+        where F: Fn(&str, &[u32]) -> bool
+    {
+        extern "C" fn foreach_cb<F>(name: *const OnigUChar, name_end: *const OnigUChar,
+            ngroup_num: c_int, group_nums: *const c_int, regex: OnigRegex, arg: *mut c_void)
```

Comment:
`Regex` is an unused parameter here. You can discard it with `_`. Probably best to do so rather than trying to pass this back the callback as callers can capture the source regex if they need it.